### PR TITLE
Add new rule for max-len to eslint configuration + fixing all errors

### LINKED ISSUE #3290: .eslintrc adding rule to wrap comment lines
True. I've tested it locally and only the lines of code are being wrapped above 80 characters. and not comments. 
I fixed to comment positioning of this PR and it is good to go. 
@shabtaisharon feel free to file a PR with your suggested fix, adding this rule might be a good addition to grommet, thanks for your inputs!

https://github.com/grommet/grommet/pull/3288

_Originally posted by @ShimiSun in https://github.com/_render_node/MDI0OlB1bGxSZXF1ZXN0UmV2aWV3Q29tbWVudDMxNjM4MjMzOA==/comments/review_comment_

### PR DESCRIPTION
**Pull request by shabtaisharon**
#### What does this PR do?
1. Added new rule for max-len to eslint configuration
1. Fixed all errors
#### Where should the reviewer start?
.eslintrc
#### What testing has been done on this PR?
storybook
grommet-site (tested locally)
#### How should this be manually tested?
storybook
#### Any background context you want to provide?
#3290 
#### What are the relevant issues?
closes #3290 
#### Screenshots (if appropriate)
N/A
#### Do the grommet docs need to be updated?
No
#### Should this PR be mentioned in the release notes?
No
#### Is this change backwards compatible or is it a breaking change?
backwards compatible

### CODE REVIEW

File: src/js/components/Diagram/stories/Animated.js [hunk 1]
```diff
@@ -9,8 +9,17 @@ import { data } from './data';
 
 const customTheme = deepMerge(grommet, {
   diagram: {
-    extend:
-      '@keyframes example { to { stroke-dashoffset: 0; } } path { stroke-dasharray: 500; stroke-dashoffset: 500; animation: example 3s linear forwards; }',
+    extend: `@keyframes 
+          example {
```
Comment (ShimiSun) [inline src/js/components/Diagram/stories/Animated.js]: The white spaces before the word 'example' are inconsistent. it should be 2 spaces.

**Comment by ShimiSun** [comment]
We've like to exclude all auto-generated files of the grommet, i.e. snapshots and readme files. will you be willing to make a further contribution?

**Comment by shabtaisharon** [comment]
> We've like to exclude all auto-generated files of the grommet, i.e. snapshots and readme files. will you be willing to make a further contribution?

Sure, I'll try to get to it in the next couple of days.

**Comment by shabtaisharon** [comment]
@ShimiSun as I understand it eslint works only on JavaScript files by default.
The reason the `READNE.md` files changes are due to changes in `doc.js` files.
Do you want me to exclude all `doc.js` files from this new rule?

**Pull request merged**